.github/workflows/ci.yml

```diff
@@ -67,8 +68,13 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           extra-packages: pytest
+          with-torch: ${{ matrix.with-torch }}
       - name: Run the unit tests & generate coverage report
+        if: ${{ matrix.with-torch == 'with-torch' }}
         run: pytest --cov --cov-fail-under=100
+      - name: Run the unit tests without torch installed
+        if: ${{ matrix.with-torch != 'with-torch' }}
+        run: pytest --cov
```

Comment:
When we test "with-torch", we are testing both with torch _and_ with numpy, so we are covering everything. Without torch, we just care whether nothing fails because we already tested everything in principle. I'm not seeing the need to complicate things further but perhaps I'm missing something

Reply:
Well, they will see incomplete coverage, yes. But note that I included `torch` in the development install, so if you are a developer you need to have it.

Reply:
Ooh I did not get that :) Then I am fine !